.github/workflows/sync-to-proxy-repo.yaml

```diff
@@ -0,0 +1,70 @@
+name: sync-to-proxy-repo
+on:
+  push:
+    branches:
+      - main
+      - latest
+  workflow_dispatch:
+    inputs:
+      env:
+        description: environment
+        default: dev
+        required: true
+jobs:
+  set_env:
+    outputs:
+      env: ${{ steps.setenv.outputs.env }}
+      dest-branch: ${{ steps.dest-branch.outputs.branch }}
+    runs-on: ubuntu-latest
+    steps:
+      - id: setenv
+        run: |
+          if test -n "${{ github.event.inputs.env }}"
+          then
+            echo "::set-output name=env::${{ github.event.inputs.env }}"
+            echo "ENV=${{ github.event.inputs.env }}" >> $GITHUB_ENV
+          fi
+          if test "${{ github.event_name }}" = 'push' -a "${{ github.ref }}" = 'refs/heads/latest'
+          then
+            echo "::set-output name=env::dev"
+            echo "ENV=dev" >> $GITHUB_ENV
+          fi
+          if test "${{ github.event_name }}" = 'push' -a "${{ github.ref }}" = 'refs/heads/main'
+          then
+            echo "::set-output name=env::prod"
+            echo "ENV=prod" >> $GITHUB_ENV
+          fi
+      - id: dest-branch
+        run: |
+          case ${{ env.ENV }} in
+            dev)
+            echo "::set-output name=branch::develop"
+            ;;
+            prod)
+            echo "::set-output name=branch::main"
+            ;;
+          esac
+  build-and-deploy:
+    needs: set_env
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          folder: docs/notebooks/
+          token: ${{ secrets.GH_PAT }}
+          repository-name: flexcompute/tidy3d-example-library
+          target-folder: notebooks
+          branch: ${{ needs.set_env.outputs.dest-branch }}
+      - run: |
+          mkdir -p build_toc
+          cp docs/examples.rst build_toc/examples.rst
+
+      - uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          folder: build_toc
+          token: ${{ secrets.GH_PAT }}
+          repository-name: flexcompute/tidy3d-example-library
```

Comment:
So this is related to the new workflow in terms of the syncing the new notebooks repo, but without affecting the existing MC setup. I understand https://github.com/flexcompute/tidy3d-example-library is what is used to host https://www.flexcompute.com/tidy3d-example-library/. This github action came from the `tidy3d-docs` repo, which was basically automatically filtering out stuff not needed to render the notebooks. I've modified it so that it gets updated in the same way in order not to break compatibility, but in reality MC could use the new development `tidy3d-notebooks` repo which is equivalent rather than this proxy repo. 

In any case, this should be an equivalent github action workflow to what existed before in the `tidy3d-docs` repo. I'm not sure which is Yongwei's github handle. Is it @magiWei maybe wants to look into it? This is related to the message I sent you a few weeks ago and just including if it's handy?

> Hi Yongwei, hope you've been well!
> I have good news that maybe you find interesting. We have [the new documentation demo up and running.](https://flexcompute-tidy3d-docs-demo.readthedocs-hosted.com/en/repo_merge_no_history/) Hope you like it. We're getting prepared to eventually implement this in the 2.6 release and  it would be great to get you onboard with the plan and solve any issues to make sure we're happy with it.
> The summary of the implementation is here:
> Currently the demonstrated demo implemented flow goes as follows:
> Development occurs in tidy3d/repo_merge_no_history. The sync-to-readthedocs-repo[ action](https://github.com/flexcompute/tidy3d/blob/repo_merge_no_history/.github/workflows/sync-to-readthedocs-repo.yaml) synchronises the main repository with flexcompute-readthedocs/tidy3d-docs-demo which can be [built and hosted by readthedocs](https://flexcompute-tidy3d-docs-demo.readthedocs-hosted.com/en/repo_merge_no_history/) without access to the flexcompute organisation. This demo is set up with the corresponding flexcompute/tidy3d-notebooks/develop, so that the docs build and pull this github submodule. This is what I aim to replicate with the main repositories & branches.
> Summary of changes:
> tidy3d consolidates docs + source for development.
> tidy3d-notebooks is the new sole directory of notebooks development. (This could replace tidy3d-example-notebooks if we wanted too)
> The easiest future for of tidy3d-docs  for everyone could be that it just becomes a mirror repo to tidy3d like in the implementation I describe above. However, because the repository had to have its directories restructured, I've done some modifications to the sync and sync-to-proxy-repo Github actions. I suspect there are likely to be transition issues in the implementation which we can solve together. It's a bit tricky to test them without running them, and it'd be great to get your input and feedback here.
> I've discussed the caveats of the implementation plan [here if you want a full explanation.](https://github.com/flexcompute/tidy3d/issues/1148#issuecomment-1834036839)